internal/witness/cmd/feeder/main.go

```diff
@@ -77,22 +80,28 @@ func main() {
 		}
 	}
 	// Now set up the log client.
-	// TODO(smeiklej): This should be optional as it can be
-	// deterministically derived from the public key.
-	if *logID == "" {
-		glog.Exit("--log_id must not be empty")
+	sha := sha256.Sum256(der)
+	b64 := base64.StdEncoding.EncodeToString(sha[:])
```

Comment:
Could use hex encoding instead and avoid the troublesome chars (there's also a "websafe" b64 encoding which uses a tweaked alphabet, but I guess that might be a little bit esoteric)?